codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/plugins/AuthSchemePlugin.kt

```diff
@@ -0,0 +1,63 @@
+package software.amazon.smithy.aws.swift.codegen.plugins
+
+import software.amazon.smithy.aws.swift.codegen.AWSClientRuntimeTypes
+import software.amazon.smithy.aws.swift.codegen.AWSSwiftDependency
+import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.swift.codegen.AuthSchemeResolverGenerator
+import software.amazon.smithy.swift.codegen.ClientRuntimeTypes
+import software.amazon.smithy.swift.codegen.SwiftWriter
+import software.amazon.smithy.swift.codegen.integration.Plugin
+import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
+import software.amazon.smithy.swift.codegen.integration.ServiceConfig
+import software.amazon.smithy.swift.codegen.model.buildSymbol
+import software.amazon.smithy.swift.codegen.model.toGeneric
+import software.amazon.smithy.swift.codegen.model.toOptional
+import software.amazon.smithy.swift.codegen.utils.toUpperCamelCase
+
+class AuthSchemePlugin(private val serviceConfig: ServiceConfig) : Plugin {
+
+    private val pluginName: String = "${serviceConfig.clientName.toUpperCamelCase()}AuthSchemePlugin"
+
+    override val className: Symbol
+        get() = buildSymbol {
+            this.name = pluginName
+        }
+
+    override val isDefault: Boolean
+        get() = true
+
+    override fun render(ctx: ProtocolGenerator.GenerationContext, writer: SwiftWriter) {
+        writer.addImport(AWSSwiftDependency.AWS_CLIENT_RUNTIME.target, false, "FileBasedConfig")
+
+        writer.openBlock("public class $pluginName: Plugin {", "}") {
+            writer.write("private var authSchemes: \$N", ClientRuntimeTypes.Auth.AuthSchemes.toOptional())
+            writer.write("private var authSchemeResolver: \$N", ClientRuntimeTypes.Auth.AuthSchemeResolver.toOptional())
+            writer.write("private var awsCredentialIdentityResolver: \$N", AWSClientRuntimeTypes.Core.AWSCredentialIdentityResolver.toGeneric().toOptional())
+
+            writer.openBlock(
+                "public init(authSchemes: \$N = nil, authSchemeResolver: \$N = nil, awsCredentialIdentityResolver: \$N = nil) {", "}",
+                ClientRuntimeTypes.Auth.AuthSchemes.toOptional(),
+                AuthSchemeResolverGenerator.getServiceSpecificAuthSchemeResolverName(ctx).toOptional(),
+                AWSClientRuntimeTypes.Core.AWSCredentialIdentityResolver.toGeneric().toOptional()
+            ) {
+                writer.write("self.authSchemeResolver = authSchemeResolver")
+                writer.write("self.authSchemes = authSchemes")
+                writer.write("self.awsCredentialIdentityResolver = awsCredentialIdentityResolver")
```

Comment:
Q: Why is credential identity resolver part of the service specific auth scheme plugin, but not the default AWS auth scheme plugin? Is this overlap between auth scheme plugin & credential identity resolver plugin intentional?

Reply:
> Is this overlap between auth scheme plugin & credential identity resolver plugin intentional?
Yes, I thought they are all related Identity and authentication so I decided to group in the same plugin.

Reply:
I see, that's fine, but should it be consistent between default plugin and service-specific plugin? Default one does not have identity resolver but service-specific one does. Also, does this make identity resolver plugin redundant?

Reply:
I think the identity resolver plugin doesn't exist in the PR anymore, but the title and description need to be updated.